source/common.ts

```diff
@@ -0,0 +1,479 @@
+    locale?: string,
+    reactjs?: typeof import('react'),
+    type?: React.ComponentType | string,
+  ) {
+    const translator = typeof translatorSeed === 'string'
+      ? i18n.createTranslator(translatorSeed, locale)
+      : typeof translatorSeed === 'undefined'
+      ? i18n.createTranslator()
+      : translatorSeed;
+
+    if (!reactjs) {
+      if (typeof React !== 'undefined') {
+        reactjs = React;
+      } else {
+        try {
+          reactjs = require('react');
+        } catch (error) {
+          // Ignore.
+        }
+      }
+
+      if (!reactjs) {
+        console.error('React is not detected!');
+      }
+    }
+
+    type Props = {
+      _containerType?: React.ComponentType | string;
+      _props?: {};
+      _tagType?: React.ComponentType | string;
+      _translateProps?: string[];
+      children?: React.ReactNode;
+    };
+
+    return class T extends reactjs!.Component<Props> {
+      static __ = translator;
+
+      _invalidate = () => this.forceUpdate();
+
+      render() {
+        const {
+          _containerType,
+          _props = {},
+          _tagType,
+          _translateProps,
+          children,
+          ...params
+        } = this.props;
+
+        const tagType = _tagType || type || 'span';
+        const items = reactjs!.Children.map(children, (item, index) => {
+          if (typeof item === 'string' || typeof item === 'number') {
+            return reactjs!.createElement(tagType, {
+              ..._props,
+              dangerouslySetInnerHTML: { __html: translator(item, params) },
+              key: `_${index}`,
+            } as any);
+          }
+
+          if (Array.isArray(_translateProps)) {
+            const newProps: Record<string, string> = {};
+            _translateProps.forEach(propName => {
+              const prop = (item as any).props[propName];
+              if (prop && typeof prop === 'string') {
+                newProps[propName] = translator(prop, params);
+              }
+            });
+
+            return reactjs!.cloneElement(item as any, newProps);
+          }
+
+          return item;
+        });
+
+        if (items?.length === 1) {
+          return items[0];
+        }
+
+        const containerType = _containerType || type || 'div';
+        return reactjs!.createElement(containerType, { ..._props }, items);
+      }
+
+      componentDidMount() {
+        i18n._events.on('changeLocale', this._invalidate);
+      }
+
+      componentWillUnmount() {
+        i18n._events.removeListener('changeLocale', this._invalidate);
+      }
+    }
+  },
+  createReactiveTranslator(namespace?: string, locale?: string) {
+    const translator = i18n.createTranslator(namespace, locale);
+    return (...args: unknown[]) => {
+      i18n._deps.depend();
+      return translator(...args);
+    };
+  },
+  createTranslator(
+    namespace?: string,
+    options?: string | CreateTranslatorOptions,
+  ) {
+    if (typeof options === 'string' && options) {
```

Comment:
Is the second truthy check necessary? The first part would already indicate that the variable is truthy. If the second check is to avoid an empty string then I think it's better to do it explicitly.